lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -1672,9 +1674,72 @@ void RemoveExplicitModules(std::vector<std::string> &args) {
 
 } // namespace
 
-void SwiftASTContext::AddExtraClangArgs(const std::vector<std::string> &ExtraArgs) {
+/// LLDB wrapper for `clang::driver::applyOverrideOptions` (which implements
+/// CCC_OVERRIDE_OPTIONS behavior).
+static void applyOverrideOptions(std::vector<std::string> &args,
+                                 llvm::StringRef overrideOpts) {
+  if (overrideOpts.empty())
+    return;
+
+  // Convert input args to the type required by applyOverrideOptions.
+  llvm::SmallVector<const char *, 64> raw_args;
+  // Add placeholder clang executable, which applyOverrideOptions expects to be
+  // the first argument.
+  raw_args.push_back("clang");
+  for (const std::string &arg : args)
+    raw_args.push_back(arg.data());
```

Comment:
I find this very scary because we don't know if args contents is nullterminated and it's easy to break this contract in the future.
Could we just make this a `std::vector<std::string>`?

Reply:
Oh, I see!
Is the null-terminator guaranteed, or should this be `arg.c_str()`?

Reply:
it's guaranteed to be null terminated:

> The returned array is null-terminated, that is, `data()` and `c_str()` perform the same function. (since C++11)

https://en.cppreference.com/w/cpp/string/basic_string/data